h/models/annotation.py

```diff
@@ -122,6 +144,16 @@ class Annotation(Base):
                                  viewonly=True,
                                  uselist=True)
 
+    @sa.orm.validates('motivations')
+    def validate_motivations(self, key, motivations):
+        # We can't use enum in an array field because SQLAlchemy doesn't
+        # support it, so we have to do validation here
+        if len(set(motivations)) != len(motivations):
+            raise ValueError(_('motivations must not contain duplicates'))
+        if len(set(motivations) - set(MOTIVATIONS)) != 0:
```

Comment:
I wonder if `MOTIVATIONS` should just be defined as a set literal (`MOTIVATIONS = {"assessing", "bookmarking", ...}`) in the first place? Then you wouldn't need to convert the list to a set here. On the other hand `MOTIVATIONS` would no longer be ordered, but it doesn't look like that matters.

Reply:
I wonder if `MOTIVATIONS` should be defined within `validate_motivations()`? Since that's the only place where it's used